pyuvdata/tests/test_mwa_corr_fits.py

```diff
@@ -294,3 +294,64 @@ def test_diff_obs():
         mwa_uv.read([bad_obs, filelist[0], filelist[1]])
     assert str(cm.value).startswith('files from different observations')
     del(mwa_uv)
+
+
+@pytest.mark.filterwarnings("ignore:telescope_location is not set. ")
+@pytest.mark.filterwarnings("ignore:coarse channels are not contiguous for this observation")
+@pytest.mark.filterwarnings("ignore:some coarse channel files were not submitted")
+def test_flag_init():
+    """
+    Test that routine MWA flagging works as intended.
+    """
+    spoof_file1 = os.path.join(DATA_PATH, 'test/spoof_01_00.fits')
+    spoof_file6 = os.path.join(DATA_PATH, 'test/spoof_06_00.fits')
+    # spoof box files of the appropriate size
+    with fits.open(filelist[1]) as mini1:
+        mini1[1].data = np.repeat(mini1[1].data, 8, axis=0)
+        extra_dat = np.copy(mini1[1].data)
+        for app_ind in range(2):
+            mini1.append(fits.ImageHDU(extra_dat))
+        mini1[2].header['MILLITIM'] = 500
+        mini1[2].header['TIME'] = mini1[1].header['TIME'] + 0.5
+        mini1[3].header['MILLITIM'] = 1000
+        mini1[3].header['TIME'] = mini1[1].header['TIME'] + 1
+        print(mini1[1].data.shape)
+        mini1.writeto(spoof_file1)
+
+    with fits.open(filelist[2]) as mini6:
+        mini6[1].data = np.repeat(mini6[1].data, 8, axis=0)
+        extra_dat = np.copy(mini6[1].data)
+        for app_ind in range(2):
+            mini6.append(fits.ImageHDU(extra_dat))
+        mini6[2].header['MILLITIM'] = 500
+        mini6[2].header['TIME'] = mini6[1].header['TIME'] + 0.5
+        mini6[3].header['MILLITIM'] = 1000
+        mini6[3].header['TIME'] = mini6[1].header['TIME'] + 1
+        mini6.writeto(spoof_file6)
+
+    flag_testfiles = [spoof_file1, spoof_file6, filelist[0]]
+
+    uv = UVData()
+    uv.read(flag_testfiles, flag_init=True, start_flag=0, end_flag=0)
+    freq_inds = [0, 1, 4, 6, 7, 8, 9, 12, 14, 15]
+
+    assert np.all(uv.flag_array[:, :, freq_inds, :]), "Not all of edge and center channels are flagged!"
```

Comment:
This ensures that the frequencies that should be flagged are flagged, but I don't see a test that ensures that frequencies that shouldn't be flagged are not flagged.

Reply:
Unfortunately this is difficult because the two test box files being read in are not contiguous in time. This means that the data_array gets filled with zeros and the flags get filled with ones where there are not actually data.

Reply:
Would it be sufficient to check that no other frequency is flagged in entirety?

Reply:
yes, I think that would address my concern.

Reply:
I've added a check that no other frequency is flagged in its entirety.